mmv1/third_party/terraform/services/compute/data_source_google_compute_reservation_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package compute_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccDataSourceComputeReservation(t *testing.T) {
+	t.Parallel()
+
+	reservationName := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+
+	rsName := "foobar"
+	rsFullName := fmt.Sprintf("google_compute_reservation.%s", rsName)
+	dsName := "my_reservation"
+	dsFullName := fmt.Sprintf("data.google_compute_reservation.%s", dsName)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckComputeReservationDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceComputeReservationConfig(reservationName, rsName, dsName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceComputeReservationCheck(t, dsFullName, rsFullName),
```

Comment:
Could you use `acctest.CheckDataSourceStateMatchesResourceState` or `CheckDataSourceStateMatchesResourceStateWithIgnores` instead of a custom function? It seems like that would miss the self link or resource name comparison (but that should be exactly the same on both since it's an output-only field) and it would miss checking the value of the status field (but that could be handled by adding `resource.TestCheckResourceAttr` for it)